client/ayon_shotgrid/plugins/publish/collect_shotgrid_session.py

```diff
@@ -17,23 +17,25 @@ def process(self, context):
                 "Have you logged in into Ayon Tray > Shotgrid ?"
             )
 
+        shotgrid_module = context.data["openPypeModules"]["shotgrid"]
+        shotgrid_url = shotgrid_module.get_sg_url()
+
         self.log.info(
-            "Attempting to create the Shotgrid Session for user: {0}".format(
-                user_login
+            "Attempting to create the Shotgrid Session for user: {0} at {1}".format(
+                user_login,
+                shotgrid_url
             )
         )
 
-        shotgrid_module = context.data["openPypeModules"]["shotgrid"]
-        shotgrid_url = shotgrid_module.get_sg_url()
-
         try:
-            sg_session = shotgrid_module.get_shotgrid_session(user_login)
+            sg_session = shotgrid_module.create_shotgrid_session(user_login)
             self.log.info("Succesfully logged in into the Shotgrid API.")
-        except Exception:
+        except Exception as e:
             raise KnownPublishError(
-                "Could not connect to Shotgrid {} with user {}".format(
+                "Could not connect to Shotgrid {0} with user {1} with error:\n{2}".format(
```

Comment:
This is message for user. Not sure how much will the error message help him? Maybe rather use log for more details and keep the message simple for user:
```
except Exception:
    self.log.error("Failed to connect to shotgrid.", exc_info=True)
    raise KnownPublishError((
            "Could not connect to Shotgrid {} with user {}"
    ).format(shotgrid_url, user_login))
```

This is based on premise that shotgrid exceptions are not "artist understandable".

Reply:
Yeah, sound better.